Libraries/src/Amazon.Lambda.AspNetCoreServer/Internal/InvokeFeatures.cs

```diff
@@ -354,6 +356,33 @@ public Task<X509Certificate2> GetClientCertificateAsync(CancellationToken cancel
 
         #endregion
 
+        #region IHttpRequestIdentifierFeature
+
+        string _traceIdentifier;
+        string IHttpRequestIdentifierFeature.TraceIdentifier
+        {
+            get 
+            {
+                if(_traceIdentifier != null)
+                {
+                    return _traceIdentifier;
+                }
+
+                var lambdaTraceId = Environment.GetEnvironmentVariable("_X_AMZN_TRACE_ID");
+                if (!string.IsNullOrEmpty(lambdaTraceId))
+                {
+                    return lambdaTraceId;
+                }
+
+                // If there is no Lambda trace id then fallback to the trace id that ASP.NET Core would have generated.
+                var baseFeature = new Microsoft.AspNetCore.Http.Features.HttpRequestIdentifierFeature();
+                return baseFeature.TraceIdentifier;

```

Comment:
Would new-ing up a new instance every time cause the TraceIdentifier value to change on each invocation if the environment variable isn't set?

Reply:
Oh, yes. Need to capture that in `_traceIdentifier`, so:
```
_traceIdentifier = baseFeature.TraceIdentifier;
return _traceIdentifier;
```

Reply:
It would change for every invocation which is the desired behavior. In the Lambda case the Lambda environment is changing the `_X_AMZN_TRACE_ID` environment variable for every invocation.

Reply:
What I mean is would it change between different accesses within the lifetime of a single "HTTP request"? I would expect it to be stable while dealing with a single invocation.

Reply:
@normj Yes, if the environment variable is set. If it is not, it will get a new string on each "get" of the `TraceIdentifier` property. I imagine that should be constant for the lifecycle of the request?

Reply:
I understand now. I was being naive thinking the getter was only called once to construct the `HttpContext` but that is probably not true and `HttpContext` forwards down into the feature set. I update the code to hold on to the identifier.